Mathlib/NumberTheory/ModularForms/JacobiTheta/OneVariable.lean

```diff
@@ -21,24 +19,49 @@ and proves the modular transformation properties `θ (τ + 2) = θ τ` and
 show that `θ` is differentiable on `ℍ`, and `θ(τ) - 1` has exponential decay as `im τ → ∞`.
 -/
 
-
-open Complex Real Asymptotics Filter
+open Complex Real Asymptotics Filter Topology
 
 open scoped Real BigOperators UpperHalfPlane
 
-/-- Jacobi's theta function `∑' (n : ℤ), exp (π * I * n ^ 2 * τ)`. -/
-noncomputable def jacobiTheta (z : ℂ) : ℂ :=
-  ∑' n : ℤ, cexp (π * I * (n : ℂ) ^ 2 * z)
+/-- Jacobi's one-variable theta function `∑' (n : ℤ), exp (π * I * n ^ 2 * τ)`. -/
+noncomputable def jacobiTheta (τ : ℂ) : ℂ := ∑' n : ℤ, cexp (π * I * (n : ℂ) ^ 2 * τ)
 #align jacobi_theta jacobiTheta
 
-theorem norm_exp_mul_sq_le {z : ℂ} (hz : 0 < z.im) (n : ℤ) :
-    ‖cexp (π * I * (n : ℂ) ^ 2 * z)‖ ≤ rexp (-π * z.im) ^ n.natAbs := by
-  let y := rexp (-π * z.im)
-  have h : y < 1 := exp_lt_one_iff.mpr (mul_neg_of_neg_of_pos (neg_lt_zero.mpr pi_pos) hz)
+lemma jacobiTheta_eq_jacobiTheta₂ (τ : ℂ) : jacobiTheta τ = jacobiTheta₂ 0 τ := tsum_congr (by simp)
+
+theorem jacobiTheta_two_add (τ : ℂ) : jacobiTheta (2 + τ) = jacobiTheta τ := by
+  simp_rw [jacobiTheta_eq_jacobiTheta₂, add_comm, jacobiTheta₂_add_right]
+#align jacobi_theta_two_add jacobiTheta_two_add
+
+theorem jacobiTheta_T_sq_smul (τ : ℍ) : jacobiTheta (ModularGroup.T ^ 2 • τ :) = jacobiTheta τ := by
```

Comment:
If you state it with `T ^ (2 : ℤ)` (and in the suffices) the  you can get rid of the `have`, although I haven't tested if itll break somewhere else...

Reply:
Unlikely. because nothing quotes this lemma yet. (I wanted to prepare the ground for proving that even powers of $\theta$ are modular forms; but that relies on the algebraic fact that the group generated by $T^2$ and $S$ contains $\Gamma(2)$, and mathlib doesn't know this, AFAIK.)

Reply:
Anyway this lemma is copied verbatim from the existing code so I'd rather leave it as it is for now.

Reply:
Ah I see, then yeah maybe leave it for now then.